File: metsis-ee-client/src/app/admin-workdesk/admin-workdesk-preparation-data.ts
import {UserMinimal} from '../user/user-minimal';

export class AdminWorkdeskPreparationData {
  public counties: any[] = [];
  public municipalities: any[] = [];
  public ownerTypes: any[] = [];
  public callers: UserMinimal[] = [];
  public statuses: any[] = [];
  public assignees: any[] = [];

  constructor(data: IAdminWorkdeskPreparationData) {
    data.counties.forEach(elem => {
      this.counties.push({id: elem, itemName: elem});
    });
    data.municipalities.forEach(elem => {
      this.municipalities.push({id: elem, itemName: elem});
    });
    data.ownerTypes.forEach(elem => {
      this.ownerTypes.push({id: elem, itemName: elem});
    });
    data.statuses.forEach(elem => {
      this.statuses.push({id: elem, itemName: elem});
    });
    data.callers.forEach(elem => {
      this.assignees.push({id: elem.id, itemName: elem.name});
    });
    this.callers = data.callers;
  }
}

export interface IAdminWorkdeskPreparationData {
  counties: string[],
  municipalities: string[],
  callers: UserMinimal[],
  ownerTypes: string[],
  statuses: string[];
}

File: metsis-ee-api/src/main/java/ee/metsis/adminworkdesk/AdminWorkdeskPreparationData.java
package ee.metsis.adminworkdesk;

import ee.finenet.fineframe.user.UserMinimal;

import java.util.List;

public class AdminWorkdeskPreparationData {
    private final List<String> counties;
    private final List<String> municipalities;
    private final List<UserMinimal> callers;
    private final List<String> statuses;
    private final List<String> ownerTypes;

    public AdminWorkdeskPreparationData(List<String> counties, List<String> municipalities, List<UserMinimal> callers, List<String> ownerTypes, List<String> statuses) {
        this.counties = counties;
        this.municipalities = municipalities;
        this.callers = callers;
        this.statuses = statuses;
        this.ownerTypes = ownerTypes;
    }

    public List<String> getCounties() {
        return counties;
    }

    public List<String> getMunicipalities() {
        return municipalities;
    }

    public List<UserMinimal> getCallers() {
        return callers;
    }

    public List<String> getStatuses() {
        return statuses;
    }

    public List<String> getOwnerTypes() {
        return ownerTypes;
    }

}
